README.md

```diff
@@ -1,7 +1,37 @@
-# rds-contracts
+# RIF Marketplace Storage Pinning
 
-## message
-The `StorageProvider` can emit an arbitrary message in the function calls `setStorageOffer` and `emitMessage`. The message is is initially intended to allow the `StorageProvider` to communicate his `nodeID`, but may be used for other purposes later.
+```
+npm i @rsksmart/rif-marketplace-storage-pinning
+```
+
+## PinningManager contract
+
+### Message
+The `StorageProvider` can emit an arbitrary message in the function calls `setStorageOffer` and `emitMessage`. 
+The message is is initially intended to allow the `StorageProvider` to communicate his `nodeID`, but may be used for other purposes later.
 
 The client should be instructed to enforce the following semantics:
-- If `message` starts with `0x01`, then the following bits contain an arbitrary length `nodeID`.
\ No newline at end of file
+- If `message` starts with `0x01`, then the following bits contain an arbitrary length `nodeID`.
+
+## Glossary
+
+ - Actors
+    - **Consumer** - actor that wants his files to be persisted in the network
+    - **Provider** - actor that offers his storage for Consumers to be used in monetary exchange
+ - Entities
+    - **Offer** - Offer created by Provider that announces availability of his storage space
+      - *Billing Period* - Specifies price for recurring period of time (e.g. 10 wei per day ) 
+      - *Maximum Duration* - Duration which Consumer can maximally rent the storage space for 
+      - *Total Capacity* - Total capacity that Provider is offering
+      - *Free Capacity* - Capacity in given time that is left free for Consumers to use  
+      - *Location* - Approximate location of the provider
+      - *Offer Termination* - Provider can choose to discontinue his Offer. No new Agreements will be accepted, but current Agreements will be finished based on their current Billing Period.
+    - **Data Source** - Data Source is an abstract entity that defines the data that should be stored under an Agreement. There might be different mechanisms on how files should be retrieved, updated and removed.
```

Comment:
We are calling it **StorageType** or **StorageSystem**, as it is something bigger than just a DataSource. Ideally I would use terminology you are handling in RIF Storage library which is "**Provider**", but the risk here is to get confused with the actual Provider of the Pinning service, so in that case would need to be renamed to **DataProvider** or something like that (Im not convinced)

Reply:
This is not a "Storage System" as you presented in the mockups. This is an abstraction on how to specify "what files to be pinned" and not "on which storage system they are stored". These two are linked as some Data Sources will be available only for some Storage Systems, but I would argue that these two things should stay separated.

I will add the "Storage System" term even though we won't use it for the first phase.

Reply:
I am missing:

> *Swarm Feeds Source* - [That is also the future babe.](https://gph.is/1FD4aQ0)